### PR TITLE
feat: support initial-response in RPC based event streams

### PR DESCRIPTION
**Pull request by dayaffe**
## Issue \#
#1163 

## Description of changes
`awaitInitialResponse()` and `retrieveInitialResponse()` are used on in combination to allow code-generated service clients to get and populate initial-response fields. Together they wait for the **first message** to be processed and then return the value of initialResponse or `nil` if it is not set.

Companion PR: [#597](https://github.com/smithy-lang/smithy-swift/pull/597)

Known Risk: We cannot currently this feature in production (integration test) as there is no known service which supports initial-response over RPC protocols.

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sichanyoo — COMMENTED** [review]
Left a question on a message encoder test case

**Review by sichanyoo — APPROVED** [review]
Lgtm!

File: Sources/Core/AWSClientRuntime/EventStream/AWSMessageDecoder.swift [hunk 1]
```diff
@@ -88,6 +103,38 @@ extension AWSEventStream {
...
+        // It uses Swift's concurrency model to asynchronously return the data.
+        public func awaitInitialResponse() async -> Data? {
+            // The 'withCheckedContinuation' function is used to bridge asynchronous code
+            // that doesn't use Swift's concurrency model with code that does.
+            return await withCheckedContinuation { continuation in
+                // Here, we attempt to retrieve the initial response.
+                // Once the data is retrieved (or determined to be nil),
+                // the continuation is resumed with the result.
+                retrieveInitialResponse { data in
+                    continuation.resume(returning: data)
```
Comment (jbelkins) [inline Sources/Core/AWSClientRuntime/EventStream/AWSMessageDecoder.swift:116]: It is required, in all cases, that the continuation gets called (it's a rather large memory leak and a frozen path of execution if it doesn't.)

Considering that `retrieveInitialResponse()` will in some cases save the continuation to an instance var, how do we guarantee that it gets called in every case?

Comment (jbelkins) [reply Sources/Core/AWSClientRuntime/EventStream/AWSMessageDecoder.swift:118]: Agreed in separate conversation that immediately after running the continuation stored in `didProcessInitialMessage`, set it to nil so it is released and it cannot be called again.

**Review by jbelkins — APPROVED** [review]

**Pull request merged**